### ISSUE TITLE
Client connection state stays as handshaking

### ISSUE THREAD

**Issue by rbirkby** [PrismarineJS/node-minecraft-protocol#187]
I can connect to a server and read chat messages posted. However, when I try to client.write, nothing happens.

I had a look at client._state and it stays as 'handshaking'.

Is this the reason I'm unable to write messages? 

Server version: 1.8.4
JavaScript RSA only.

**Comment by rom1504** [comment]
Can you post the code you tried ?

**Comment by rbirkby** [comment]
```javascript
var mc = require('minecraft-protocol');
var client = mc.createClient({
  host: "{1.8.4 server}",
  username: "{username}",
  password: "{password}",
});
client.on('chat', function(packet) {
    client.write(0x03, {
      message: 'hello world'
    });
});
```

**Comment by roblabla** [comment]
._. you shouldn't even be able to read the messages if the client's state is stuck at HANDSHAKING. What version of minecraft-protocol are you using ? (NPM version/github commit).

Also, which server did you try to connect to exactly (ip address or software+version) ?

**Comment by rbirkby** [comment]
"minecraft-protocol": "^0.14.0"

The server is one I spun up in DigitalOcean a while ago. It's running 1.8.4. mc.ping says:

  version: { name: '1.8.4', protocol: 47 },

**Comment by rom1504** [comment]
0x03 is not the chat packet. You can write 'chat' instead of the packet number.

**Comment by rbirkby** [comment]
I was going on the docs:

https://www.npmjs.com/package/minecraft-protocol#echo-client-example

**Comment by roblabla** [comment]
@rbirkby does using 'chat' instead of 0x03 fix your issue ?

**Comment by rbirkby** [comment]
events.js:85
	      throw er; // Unhandled 'error' event
	            ^
	Error: ForbiddenOperationException
	    at C:\src\MinecraftBot\node_modules\minecraft-protocol\dist\yggdrasil.js:48:21
	    at Request.callback (C:\src\MinecraftBot\node_modules\minecraft-protocol\node_modules\superagent\lib\node\index.js:575:3)
	    at Request.<anonymous> (C:\src\MinecraftBot\node_modules\minecraft-protocol\node_modules\superagent\lib\node\index.js:133:10)
	    at Request.emit (events.js:107:17)
	    at IncomingMessage.<anonymous> (C:\src\MinecraftBot\node_modules\minecraft-protocol\node_modules\superagent\lib\node\index.js:703:12)
	    at IncomingMessage.emit (events.js:129:20)
	    at _stream_readable.js:908:16
	    at process._tickCallback (node.js:355:11)


```javascript
client.on('chat', function() {
	client.write('chat', {message:'hello world'});
});
```

Alternatively without the client.on() handler:


    C:\src\MinecraftBot\node_modules\minecraft-protocol\dist\client.js:207
    iting packetId ' + this.state + '.' + packetName + ' (0x' + packetId.toString(
                                                                        ^
    TypeError: Cannot read property 'toString' of undefined
        at Client.write (C:\src\MinecraftBot\node_modules\minecraft-protocol\dist\client.js:207:80)
        at Object.<anonymous> (C:\src\MinecraftBot\index.js:8:8)
        at Module._compile (module.js:460:26)
        at Object.Module._extensions..js (module.js:478:10)
        at Module.load (module.js:355:32)
        at Function.Module._load (module.js:310:12)
        at Function.Module.runMain (module.js:501:10)
        at startup (node.js:129:16)
        at node.js:814:3

**Comment by rom1504** [comment]
I think we can close this since #195 is fixed by #197
